common/src/main/java/com/viaversion/viabackwards/protocol/v1_20_5to1_20_3/rewriter/EntityPacketRewriter1_20_5.java

```diff
@@ -345,6 +355,20 @@ private void updateDimensionTypeData(final CompoundTag elementTag) {
         }
     }
 
+    private void sendCarpetColorUpdate(final UserConnection connection, final int entityId, final Item item) {
+        final PacketWrapper setEntityData = PacketWrapper.create(ClientboundPackets1_20_3.SET_ENTITY_DATA, connection);
+        setEntityData.write(Types.VAR_INT, entityId); // Entity id
+        int color = -1;
+        if (item != null) {
+            // Convert carpet item id to dyed color id
+            if (item.identifier() >= 445 && item.identifier() <= 460) {
+                color = item.identifier() - 445;
+            }
+        }
+        setEntityData.write(Types1_20_3.ENTITY_DATA_LIST, List.of(new EntityData(20, Types1_20_3.ENTITY_DATA_TYPES.varIntType, color)));
```

Comment:
Should be a mutable list

Reply:
array list instead of linked list, event handlers use indexes and other data might be added

Reply:
I saw another packet using it when sending new entity data, but fixed